clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -4336,4 +4384,9 @@ public boolean hasAllPartitionReportedCompleted() {
   void setVersionRole(PartitionReplicaIngestionContext.VersionRole versionRole) {
     this.versionRole = versionRole;
   }
+
+  // Only for testing purpose.
```

Comment:
I understand everyone has their own habit. May I suggest to still make the divChunkAssembler final but creating a constructor that allows you to pass a mock divChunkAseembler? In that way, you don't have to have such a setter. 

Either way needs one more method but the "setting a variable final" is better in execution for performance. Just a bit.

Reply:
SG. I will fix it.